gen/python/grpc/test/benchmark_test.py

```diff
@@ -0,0 +1,177 @@
+import json
+import random
+import uuid
+from typing import List
+from google.protobuf.json_format import MessageToDict
+
+import pytest
+
+from caraml.upi.v1 import upi_pb2, value_pb2
+
+# represents number of prediction rows x number of prediction values
+bench_response_test_cases = [
+    (1, 1),
+    (1, 10),
+    (100, 1),
+    (100, 10),
+    (1000, 1),
+    (1000, 10),
+]
+
+# represents number of rows x number of featuress
+bench_request_test_cases = [
+    (1, 1),
+    (1, 100),
+    (100, 100),
+    (100, 500),
+    (1000, 100),
+    (1000, 500),
+]
+
+
+@pytest.mark.parametrize("n_rows,n_cols", bench_request_test_cases)
+@pytest.mark.benchmark(group="serialize-request")
+def test_serialize_proto_request(n_rows, n_cols, benchmark):
+    request = create_request(n_rows, n_cols, )
+    benchmark(serialize_request, request=request)
+    benchmark.extra_info["message_size"] = request.ByteSize()
```

Comment:
where can this message size be found? cant find it anywhere

Reply:
You can only access it in the json report generated by the test. It's limitation of pytest-benchmark. I am exploring how can I expose it without having developer to clone and run the benchmark themselves.